server/workflow/store/sqlite_store.go

```diff
@@ -0,0 +1,421 @@
+package store
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/upper/db/v4"
+	"github.com/upper/db/v4/adapter/sqlite"
```

Comment:
Using `"github.com/upper/db/v4/adapter/sqlite"` will include `github.com/mattn/go-sqlite3` as a dependency which requires cgo, so I have to set `CGO_ENABLED=1` and install `build-base` in the builder docker image for argocli, also I feel like this might break cross-platform compilation.

One way to avoid this is to use some cgo-free sqlite lib, for example, https://github.com/zombiezen/go-sqlite, but I am unsure if there is a better way of handling it.

Reply:
Theres also https://gitlab.com/cznic/sqlite if you want a `database/sql` driver.

Reply:
I managed to implement the same by using the https://github.com/zombiezen/go-sqlite, and now it is cgo-free.

Reply:
> Theres also https://gitlab.com/cznic/sqlite if you want a `database/sql` driver.

> I managed to implement the same by using the https://github.com/zombiezen/go-sqlite

`zombiezen/go-sqlite` uses `cznic/sqlite` under the hood, for reference.

Also TIL about https://gitlab.com/cznic/ccgo, a C to Go compiler 😮 🤯